src/jit/MemoryInl.h

```diff
@@ -95,6 +98,12 @@ void MemAddress::check(sljit_compiler* compiler, Operand* offsetOperand, sljit_u
             memArg.arg = SLJIT_MEM1(baseReg);
             memArg.argw = offset;
             load(compiler);
+
+#if defined(ENABLE_EXTENDED_FEATURES)
+            if (size > 1 && (options & CheckNaturalAlignment) && (memArg.argw & (size - 1)) != 0) {
+                context->appendTrapJump(ExecutionContext::UnalignedAtomicError, sljit_emit_jump(compiler, SLJIT_NOT_EQUAL));
+            }
+#endif /* ENABLE_EXTENDED_FEATURES */
```

Comment:
This should be done after `if (UNLIKELY(offset > context->maximumMemorySize - size)) {`, and emit no instruction if there is an error. Maybe the 'if' part can be combined.

Reply:
How could the two `if`s be combined? the alignment check is guarded with `ENABLE_EXTENDED_FEATURES` while the other `if` is always included.

Reply:
Ok, that is ugly. Then duplicate the `if` body. I hope the compiler notices that.